src/app/Http/Traits/SignInitDocumentSignatureTrait.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+namespace App\Http\Traits;
+
+use App\Enums\BsreStatusTypeEnum;
+use App\Enums\MediumTypeEnum;
+use App\Enums\SignatureMethodTypeEnum;
+use App\Enums\SignatureQueueTypeEnum;
+use App\Enums\SignatureStatusTypeEnum;
+use App\Exceptions\CustomException;
+use App\Jobs\ProcessMultipleEsignDocument;
+use App\Models\DocumentSignature;
+use App\Models\DocumentSignatureSent;
+use Illuminate\Support\Facades\Redis;
+
+/**
+ * Setup configuration for signature document
+ */
+trait SignInitDocumentSignatureTrait
+{
+    use SignActionDocumentSignatureTrait;
+
+    public function setupSingleFileEsignDocument($requestInput, $userId = null)
+    {
+        $checkAlreadySigned = $this->checkSingleFileAlreadySigned($requestInput);
+        if ($checkAlreadySigned !== true) {
+            return $checkAlreadySigned;
+        }
+        return $this->doSingleFileEsignDocument($requestInput, $userId);
+    }
+
+    protected function checkSingleFileAlreadySigned($requestInput)
+    {
+        if ($requestInput['isSignedSelf'] == true) {
+            $documentSignature = DocumentSignature::findOrFail($requestInput['id']);
+            return $this->checkErrorSingleFileAlreadySignedSelf($documentSignature);
+        }
+
+        if ($requestInput['isSignedSelf'] == false) {
+            $documentSignatureSent = DocumentSignatureSent::findOrFail($requestInput['id']);
+            return $this->checkErrorSingleFileAlreadySigned($documentSignatureSent);
+        }
+    }
+
+    protected function checkErrorSingleFileAlreadySignedSelf($documentSignature)
+    {
+        $logData = $this->setKafkaDocumentApproveResponse($documentSignature->id);
+        if ($documentSignature->status != SignatureStatusTypeEnum::WAITING()->value && $documentSignature->is_signed_self == false) {
```

Comment:
Does this need to be fixed? How about this ?

```suggestion
        if ($documentSignature->status == SignatureStatusTypeEnum::WAITING()->value && $documentSignature->is_signed_self == true) {
```